src/core/devicefactory.h

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Martin Erich Jobst
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Martin Erich Jobst - initial implementation
+ *******************************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <string_view>
+#include <concepts>
+
+#include "core/stringdict.h"
+
+#include <unordered_map>
+
+class CDevice;
+
+namespace forte::core {
+
+  class DeviceEntry;
+
+  class DeviceFactory final {
```

Comment:
This is a class with just static members. Maybe it could be jsut a namespace, and you won't need the friends in it or in the DeviceEntry